app/views/admin/products_v3/index.html.haml

```diff
@@ -7,3 +7,9 @@
         = button_link_to t(:new_product), "/admin/products/new", { :icon => 'icon-plus', :id => 'admin_new_product' }
 
 = render partial: 'spree/admin/shared/product_sub_menu'
+
+#products_v3_page{"data-controller": "productsV3"}
+  #loading-spinner.spinner-container{"data-productsV3-target": "loading"}
+    .spinner
+    = t('.loading')
+  #products-content
```

Comment:
What happened to your opinion that a page should be progressively enhanced by JS? Shouldn't the products be rendered straight away and only be updated by a reflex?

Reply:
Good question! This is how it was specified in #10679.
I have to admit I hadn't thought through this one. while I prepared the PR it was a combination of both JB and I's work which I didn't think to review myself. 

Thinking about it, I agree. Perhaps if it was expensive data to load, we might want to load the rest of the interface first  (ie the search and filter selection) before loading the data. But with pagination that shouldn't be a problem. 
Also in a more general way, we seek to provide a responsive experience with Turbo. 

So.. yeah I'm not sure why we need the loading spinner for the initial load. We do however intend to have it for when you search/filter/paginate.

@jibees, what do you think?

Reply:
I added a loading spinner as mentionned in the spec. I think it's fine to add such a thing, since once you click on "Search" button, or pagination, interface is blocked by this spinner, and then you cannot click twice. And sometimes yes, it can take some times.

Actually, not sure to follow:
> Shouldn't the products be rendered straight away and only be updated by a reflex?

What do you mean?

Reply:
I mean that the initial page render would not show "Loading" but render the products within the document. The "Loading" would only be used for updates when you sear etc.

Reply:
I personaly prefer to display the "Loading" at startup (as well as updates), just in order to be consistent. But it's not a strong opinion.